cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/indexer/IndexerFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.indexer;
+
+import com.google.common.base.Throwables;
+
+/**
+ * A factory for {@link Indexer}
+ */
+public class IndexerFactory {
```

Comment:
Since there is no special logic necessary to instantiate an indexer, there is no need for factory, right? We can just new the classes to instantiate them. I don't see the reason for a factory, nor the reason for `IndexerType` enum. Let's keep implementation simple as far as possible.